project/apps/web/components/DifficultyBadge.tsx

```diff
@@ -0,0 +1,34 @@
+import { Badge } from "@/components/ui/badge";
+import { COMPLEXITY } from "@/constants/question";
+import { cn } from "@/lib/utils";
+
+interface DifficultyBadgeProps {
+  complexity: string;
+  className?: string;
+}
+
+const DifficultyBadge = ({ complexity, className }: DifficultyBadgeProps) => {
+  let colorClass = "";
+
+  switch (complexity) {
+    case COMPLEXITY.Easy:
+      colorClass = "bg-green-400 text-white hover:bg-green-400";
+      break;
+    case COMPLEXITY.Medium:
+      colorClass = "bg-yellow-400 text-white hover:bg-yellow-400";
+      break;
+    case COMPLEXITY.Hard:
+      colorClass = "bg-red-400 text-white hover:bg-red-400";
+      break;
+    default:
+      colorClass = "bg-secondary text-white";
+  }
```

Comment:
should we abstract this out into a separate function instead? should try to avoid using `let` as much as we can.
```
function getColorClassByComplexity(complexity: COMPLEXITY): string {
  switch (complexity) {
    case COMPLEXITY.Easy:
      return "bg-green-400 text-white hover:bg-green-400";
    case COMPLEXITY.Medium:
      return "bg-yellow-400 text-white hover:bg-yellow-400";
    case COMPLEXITY.Hard:
      return "bg-red-400 text-white hover:bg-red-400";
    default:
      return "bg-secondary text-white";
  }
}
```